src/main/java/games/strategy/ui/SwingComponents.java

```diff
@@ -66,15 +66,15 @@
    * Convenience method to create {@code AbstractAction} instances using lambdas.
    * Example usage:
    * <code><pre>
-   *   AbstractAction action = SwingComponents.newAbstractionAction(() -> {
+   *   AbstractAction action = SwingComponents.newAbstractAction("Something", () -> {
    *    if (value) {
    *      doSomething();
    *    }
    *   });
    * </pre></code>
    */
-  public static AbstractAction newAbstractAction(final Runnable action) {
-    return new AbstractAction() {
+  public static AbstractAction newAbstractAction(final String name, final Runnable action) {
```

Comment:
When using the `JButton(Action)` constructor, as we do in `AbstractMovePanel`, the button text is initialized using the `Action` name.  Because we were not setting either the button text or the action name previously, the button ended up with an empty label.

`newAbstractAction()` was added as part of #2137, so I assumed there was a purpose for it. :smile:  Maybe it was added in a commit before the `swinglib` stuff started to take shape?  IDK, but its only caller is the one line in `AbstractMovePanel`, so, at worst, we can inline it there.